### PR TITLE
Migrate DataSource vector and index map to a single UUID:DataSource map

### PR DESCRIPTION
**Pull request by greenscientist**
Added == and != operators to DataSource

Since Person and OdTrip now store a reference to a DataSource, we added constructor for those.

While doing this, we observed that Households and Places are not used anywhere in the calculation.
We have commented out those files, references and initialization functions. Those were marked
with a TODO #167 (task number to restore them when needed/refactoring done)

Tested with Transition

### CODE REVIEW

File: include/parameters.hpp [hunk 1]
```diff
@@ -149,7 +151,8 @@ namespace TrRouting
       int routingDateYear;   // not implemented, use onlyServicesIdx or exceptServicesIdx for now
       int routingDateMonth;  // not implemented, use onlyServicesIdx or exceptServicesIdx for now
       int routingDateDay;    // not implemented, use onlyServicesIdx or exceptServicesIdx for now
-      int onlyDataSourceIdx;
+    //TODO Make it a reference or not??
+    std::optional<DataSource> onlyDataSource;
```
Comment (tahini) [inline include/parameters.hpp]: check alignments in this file

File: src/data_sources_cache_fetcher.cpp [hunk 2]
```diff
@@ -67,28 +65,28 @@ namespace TrRouting
...
+          case dataSource::DataSource::Type::BIKE_SHARING_TRANSACTIONS: t.type = "bikeSharingTransactions"; break;
+          case dataSource::DataSource::Type::GPS_TRACES               : t.type = "gpsTraces";               break;
+          case dataSource::DataSource::Type::STREET_SEGMENT_SPEEDS    : t.type = "streetSegmentSpeeds";     break;
+          case dataSource::DataSource::Type::UNKNOWN                  : t.type = "unknown";                 break;
         }
 
-        tIndexesByUuid[t->uuid] = ts.size();
-        ts.push_back(std::move(t));
+        //tIndexesByUuid[t.uuid] = ts.size();
+        ts[t.uuid] = t;
```
Comment (tahini) [inline src/data_sources_cache_fetcher.cpp:89]: So there is no unique_ptr anymore, so iiuc now we have a map of objects and the `at` method of the map returns a & to the object? This line here copies the entire object? Do we have/need a copy constructor? Has it been tested?

**Review by tahini — CHANGES_REQUESTED** [review]
A c++ question and alignment issues

File: src/data_sources_cache_fetcher.cpp (hunk 2 above)
Comment (greenscientist) [reply src/data_sources_cache_fetcher.cpp:89]: Yes, it copies the whole object. 
We don't need an explicit copy constructor, the default one will do the job. 

As for testing, it seems to work with Transition.

File: include/parameters.hpp (hunk 1 above)
Comment (greenscientist) [reply include/parameters.hpp]: updated

**Comment by greenscientist** [comment]
(and rebased)

**Review by tahini — APPROVED** [review]
Looks ok, I'm really really looking forward for multi-threaded trRouting, so by all means, go ahead!!

**Comment by greenscientist** [comment]
> Looks ok, I'm really really looking forward for multi-threaded trRouting, so by all means, go ahead!!

still a long way to go :/

**Pull request merged**